sanic/app.py

```diff
@@ -309,6 +311,28 @@ def _apply_middleware(
                 middleware.middleware, middleware.attach_to
             )
 
+    def _apply_signal(self, signal: FutureSignal) -> Signal:
+        return self.signal_router.add(*signal)
+
+    def dispatch(
+        self,
+        event: str,
+        *,
+        where: Optional[Dict[str, str]] = None,
+        context: Optional[Dict[str, Any]] = None,
+    ) -> Coroutine[Any, Any, Awaitable[Any]]:
+        return self.signal_router.dispatch(
+            event,
+            context=context,
+            where=where,
+        )
+
+    def event(self, event: str):
+        signal = self.signal_router.name_index.get(event)
+        if not signal:
+            raise NotFound("Could not find signal %s" % event)
+        return signal.ctx.event.wait()
```

Comment:
Can we have the ability to add a timeout to Event?
```suggestion
    def event(self, event: str, timeout: Optional[Union[Int,Float]]):
        signal = self.signal_router.name_index.get(event)
        if not signal:
            raise NotFound("Could not find signal %s" % event)
        return asyncio.wait_for(signal.ctx.event.wait(), timeout=timeout)
```

Reply:
How does `trio` play with `wait_for`?

Reply:
I haven't tested it on trio, but I expect it to work the same as uvloop and asyncio.